rpc/notification.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+package rpc
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/logger"
+	"github.com/ethereum/go-ethereum/logger/glog"
+)
+
+var (
+	NotificationsUnsupportedErr = errors.New("notifications not supported")
+	NotificationNotFoundErr     = errors.New("notification not found")
+	notifierStoppedErr          = errors.New("unable to send notification")
+	notificationQueueFullErr    = errors.New("too many pending notifications")
+)
+
+var unsubSignal struct{}
```

Comment:
Note that all values of type `struct{}` are equal, so this doesn't serve as a unique ID.